### PR TITLE
[#6023] Test explorer should not start Azure emulator when loading tests

### LINKED ISSUE #6023: Test explorer should not start Azure emulator when loading tests
## Summary

There is some code in the Azure tests that is causing the azure emulators to startup when opening test explorer in visual studio gets in the way when testing other things that are not related to azure.

This is probably caused by some static constructor that gets triggered when test explorer uses reflection to find tests in the target assemblies. 

As part of this issue we need to refactor the code so emulator is only started when the tests are executed and, if possible, make sure they only get started once per test run and not multiple times (right now the security window pops out at least twice).

While refactoring these test, we should analyze if we really need to run azure emulator for unit tests, it may make more sense to move the tests that depend on emulator to functional and integration tests.

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes #6023

## Description
This PR disables the CosmosDB tests from running by skipping them in the `IgnoreOnNoEmulatorFact` class. It also adds a description explaining the reasons why they are being disabled.

## Specific Changes
- Removed `HasEmulator` lazy variable from `CosmosDbPartitionStorageFixture`, `CosmosDbStorageTests` and `IgnoreOnNoEmulatorFact` to prevent the Emulator security window to appear.
- Updated the `IgnoreOnNoEmulatorFact` skip message with information related to why the tests are skipped and where the tests are going to be moved.

## Testing
The following image shows the CosmosDB tests being always skipped.
imagen

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbFixture.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Bot.Builder.Azure.Tests
+{
+    public class CosmosDbFixture : IAsyncLifetime
+    {
+        // Endpoint and Authkey for the CosmosDB Emulator running locally
```
Comment (gabog) [inline tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbFixture.cs:14]:
```suggestion
        // Endpoint and Authkey for the CosmosDB Emulator running locally (see https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=ssl-netstd21#authenticate-requests for details on the well know key being used)
```

**Comment by gabog** [comment]
Hi @sw-joelmut, I tried to test this locally and on the build server and I'm not sure I understand how is supposed to work. I pulled the latest, went to test explorer and tried to run the azure tests. 
1. Loading the tests in test explorer doesn't trigger emulator (this is good)
2. When I tried running the CosmosDbPartitionStorageTests and CosmosDbStorageTests
image
   I got a security window asking me to allow running emulator (emulator wasn't running so I guess that's OK): 
image
   I said yes but the tests were skipped.

3. I manually started emulator from outside VS and tried re-running the tests, I got a security window again (not sure why because emulator was already running) and the tests executed this time. Am I supposed to manually start cosmosDb emulator before running the tests? 

4. On the build server, I downloaded the trx files for the tests run as part of the PR verification and it seems, the tests are being skipped there, do we need to add a step to start the emulator in the build server? 
image

**Comment by gabog** [comment]
Looking at the code, I realized that the security window I am getting is because is running `CosmosDB.Emulator.exe /GetStatus`. 

I think it would be better to fail the test if emulator is not running instead of skipping them. I think that IF the tests are silently being skipped on the build server then failing the tests would reveal that. Ping me tomorrow and we can chat more.

**Comment by sw-joelmut** [comment]
@gabog The tests have been disabled, a description has been added with the information why they are disabled and where they are going to be moved. Also updated the PR description for more info. 

Thanks.

**Review by gabog — APPROVED** [review]

**Pull request merged**